lib/css/CssModulesPlugin.js

```diff
@@ -501,6 +519,21 @@ class CssModulesPlugin {
 					codeGenResult.sources.get("css") ||
 					codeGenResult.sources.get("css-import");
 
+				const moduleSourceCode = moduleSource.source().toString();
+				const publicPathRegex = new RegExp(
+					CssUrlDependency.PUBLIC_PATH_PLACEHOLDER,
```

Comment:
It breaks source maps, we should not do it, we should rewrite assets modules and generate the valid URL there

Reply:
Yes, we use the same logic in mini-css-extract-plugin, but it was wrong and yeah our source maps are broken in some cases, there are issues about it

Reply:
Is there any link about the source maps issues, would like to learn more.

Reply:
Just make an URL with a newline and put somethings CSS on the same line.

Also if you look your bundled JS file, it will contain asset module, this is wrong, so the solution is refactor assets module and make it compatibility with CSS

Reply:
Good to know, thanks